### PR TITLE
Ability For Ghosts To Teleport To Players

### PR DESCRIPTION
**Pull request by ThatDan123**
Lets Ghosts teleport to:

Alive Players
Dead Bodies
Other Ghosts

100% Done, thanks Jibby

~90% done, Have UI and the Generating of the teleportable objects list.~

~10% left is the actual teleporting, the hardest part.~
~Need some help:~

~Currently cant get the server to teleport correctly as you get pinged backwards.~

~Need a setPos method to call safely on server and that doesnt use Player moving scripts.~

- [x] Code is sufficiently commented
- [x] Code is indented with tabs and not spaces
- [x] The PR does not include any unnecessary .meta, .prefab or **.unity (scene) changes**
- [x] The PR does not bring up any new compile errors
- [x] The PR has been tested in editor
- [x] The PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### CODE REVIEW

**Comment by ThatDan123** [comment]
Ready to merge as soon as testing is done!!!!!!!

**Comment by Jibbajabbafic** [comment]
Lol @ that commit message, it was nothing man

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 1]
```diff
@@ -718,8 +718,20 @@ public void CmdRequestDisarm(GameObject playerToDisarm)
 		}
 	}
 
-	//admin only commands
+	[Command]
+	public void CmdGhostPerformTeleport(Vector3 s3, GameObject localPlayer)
+	{
+		ServerGhostPerformTeleport(s3, localPlayer);
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: Pretty unsecure anyone could call cmd from their client and teleport where ever they want. I'd perform a check here to see if this player is actually a ghost before allowing them to teleport

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 2]
```diff
@@ -718,8 +718,20 @@ public void CmdRequestDisarm(GameObject playerToDisarm)
...
+	}
+
+	[Server]
+	public void ServerGhostPerformTeleport(Vector3 s3, GameObject localPlayer)
+	{
+		Logger.Log("tried to teleport to " + s3);
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: You're just going to spam useless logs on the server with this. I'd say remove it all together

**Review by DooblyNoobly — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs (hunk 2 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: Yeah sorry missed that one

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: Cool

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: `if (playerScript.IsGhost)
		{
			playerScript.PlayerSync.SetPosition(s3); //server forces position on player
		}`

**Pull request merged**